plugins/user-settings/api-report.md

```diff
@@ -42,6 +44,32 @@ export const Router: (props: { providerSettings?: JSX.Element }) => JSX.Element;
 // @public (undocumented)
 export const Settings: (props: { icon?: IconComponent }) => JSX.Element;
 
+// @public (undocumented)
+export const SettingsLayout: {
```

Comment:
Gotta say that the `SettingsLayout` vs `UserSettingsTab` does end up being a bit awkward. I wonder if we should for example aim to deprecate `UserSettingsTab` and make it possible to use `SettingsLayout.Route` directly instead?

Another option is to just completely deprecate `UserSettingsTab` and the tab-only use-case, and leave `SettingsLayout` as the only point of customization. It's a bit more verbose if you just wanna add tabs, but the API is a quite a bit cleaner.

Reply:
I'd say let's try the direct `SettingsLayout.Route` instead. So you could either use it directly to add tabs, or wrap it in the layout to completely replace all tabs. Behaving just like it does right now but using `SettingsLayout.Route` rather than `UserSettingsTab`.

This is a pretty popular plugin that's been somewhat stable for a while, so let's provide a nice deprecation path. It's fine to switch out the component data key as there's a guarantee that we're only using a single version of the package in the app. With that I'm thinking that we simply deprecate `UserSettingsTab` and set the component data key to be the same as the new `SettingsLayout.Route`, since the props are compatible. With that `USER_SETTINGS_TAB_KEY` should be deprecated as well, tbh it shouldn't have been exported in the first place.

Reply:
I've removed `UserSettingsTab` and replaced it with `SettingsLayout.Route`, but now the `SettingsPage` got more boilerplatey - I can't think of a way to check if an outlet is a `Layout` or `Route`, without checking for both, since this will return the tabs in both cases:

```tsx
const tabs = useElementFilter(outlet, elements =>
    elements
      .selectByComponentData({
        key: LAYOUT_ROUTE_DATA_KEY,
      })
      .getElements<SettingsLayoutRouteProps>(),
  );
```

If there's a better way than to use two `useElementFilter`s - please let me know

Reply:
Yeah that's definitely tricky. I think it's good as is, let's leave it as a future thing to figure out.